skimage/draw/draw.py

```diff
@@ -693,3 +693,75 @@ def bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
     return _bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape)
+
+
+def rectangle(start, end=None, extent=None, shape=None):
+    """Generate coordinates of pixels within an nd-rectangle. Either
+    end or extent are required.
+
+    Parameters
+    ----------
+    start : tuple
+        Origin point of the rectangle, ([plane], row, column[, ...])
+    end : tuple
+        End point of the rectangle ([num_planes], num_rows, num_cols[, ...]).
```

Comment:
@stefanv we haven't yet standardised on this. I would say `([plane, ]row, column[, ...])` because I'm pedantic, or even `([..., [plane, ]]row, column)`, but we have the ... on the end because you could do a rectangle along the channels... But perhaps we could remove them. Feel free to suggest alternatives! ;)